cmd/repo-query/repo-query.go

```diff
@@ -0,0 +1,353 @@
+package repoquery
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const repoQueryURL string = "https://opensauced.tools"
+
+type Options struct {
+	// URL is the git repo URL that will be indexed
+	URL string
+
+	branch string
+}
+
+const repoQueryLongDesc string = `WARNING: Proof of concept feature.
+
+The repo-query command takes a URL to a git repository and indexes it so
+users can ask questions about it.`
+
+func NewRepoQueryCommand() *cobra.Command {
+	opts := &Options{}
+
+	cmd := &cobra.Command{
+		Use:   "repo-query url [flags]",
+		Short: "Ask questions about a git repository",
+		Long:  repoQueryLongDesc,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 1 {
+				return errors.New("only a single url can be ingested at a time")
+			}
+			if len(args) == 0 {
+				return errors.New("must specify the URL of a git repository")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.URL = args[0]
+			return run(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.branch, "branch", "b", "HEAD", "The branch to index")
+
+	return cmd
+}
+
+func getOwnerAndRepo(url string) (owner, repo string) {
+	// Remove the "https://github.com/" prefix from the URL
+	url = strings.TrimPrefix(url, "https://github.com/")
```

Comment:
This is abit brittle. Entering anything other the `https://github.com/...` produces a confusing error:

```
❯ go run main.go repo-query github.com/open-sauced/pizza
Checking if github.com/open-sauced is indexed by us...⏳
Repo not found❗
Indexing repo...⏳
An error occurred: Unable to fetch repository license
Error: error while indexing repository
Usage:
  pizza repo-query url [flags]

Flags:
  -b, --branch string   The branch to index (default "HEAD")
  -h, --help            help for repo-query

2023/08/01 15:37:03 error while indexing repository
exit status 1
```

We should maybe look for a good regex that captures most requests to GitHub. We should probably also add that only github.com is supported (for now) in the help message.